pkg/kubelet/prober/worker.go

```diff
@@ -126,12 +125,10 @@ func newWorker(
 
 // run periodically probes the container.
 func (w *worker) run() {
-	probeTickerPeriod := time.Duration(w.spec.PeriodSeconds) * time.Second
-
-	// If kubelet restarted the probes could be started in rapid succession.
```

Comment:
@alanorwick thanks for the more detailed explanation in the pr :)

Could you explain more about how, after your proposed change, we have protection against the kubelet restarting and the probes being started in rapid succession? Are we relying upon there being variation in `InitialDelaySeconds`? If so, can we feel confident that assumption is safe?

Reply:
One (not fully thought out idea) that I have... could we define an "acceptable range" in which our initial sleep period must fall, and then choose a random value within that acceptable range? I believe that would allow us both to ensure we were seeing predictable results via kubectl and also not have the probes started in too rapid a succession after kubelet restart. Very possible I'm missing a reason why the above idea won't work though :)

Reply:
Hello! I am the other student working with @alanorwick to complete this PR. Is the problem with the probes being started in rapid succession that there might be an initial large stress on the worker node when probing all the pods at the same time (after a kubelet restarts)?

With our proposed change, if the kubelet restarts, all the probes will be started after the `InitialDelaySeconds` specified in their pod deployment. If `InitialDelaySeconds` is not specified in the pod deployment file, the prober will run immediately after the kubelet restarts (default of `InitialDelaySeconds` is 0). Either way, it would not be safe to assume that the different probers will start at staggered times.

When you say an "acceptable range", did you mean that users would be limited to a certain range to choose their `InitidalDelaySeconds`? Or did you mean that the user can still specify any `InitialDelaySeconds` in their pod deployment, but we would create a random range around that specification to choose the actual initial delay? Just wanted to make sure I understood your suggestion. Let me know!

Reply:
Hi! Very cool that you two are working on this for school :)

I think the most important question here is whether we actually need the jitter (and also how "large" of a jitter we actually need). https://github.com/kubernetes/kubernetes/pull/20726 is the pr which introduced the jitter. @ingvagabund I see you authored the original commit - do you mind weighing in here? Thanks :)

Reply:
That's what I was wondering as well! If we decide we need the jitter, what are your thoughts on defining a jitter range around the user specified `InitialDelaySeconds` (with a minimum of 0 seconds). This would allow users to continue as expected as defined in the documentation, and minimize the chance of a spike in CPU usage from probes starting in quick succession after a kubelet restart. @mattjmcnaughton @ingvagabund

Reply:
It's been a while. Though, I agree to limit the maximal value of `rand.Float64() * float64(probeTickerPeriod)` in case it goes too high.

Reply:
The issue is that the jitter using `probeTickerPeriod` is causing unexpected behavior when users specify `InitialDelaySeconds`, as the jitter is currently not based on `InitialDelaySeconds`. What are your thoughts on creating a random jitter time range around the user specified `InitialDelaySeconds` to both support expected behavior on the user's end and randomness in probe start up in the case that a kubelet restarts?